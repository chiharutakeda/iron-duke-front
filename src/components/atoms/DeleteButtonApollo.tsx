import { useDeleteToDoMutation } from '@/generated/graphql';
import styles from './styles/DeleteButton.module.css';

const DeleteButtonApollo = (deleteId: { ID: number }) => {
  const [deleteToDoMutation] = useDeleteToDoMutation({
    variables: {
      deleteTODO: deleteId,
    },
  });
  return (
    <button className={styles.DeleteButton} onClick={() => deleteToDoMutation()}>
      Delete
    </button>
  );
};

export default DeleteButtonApollo;
