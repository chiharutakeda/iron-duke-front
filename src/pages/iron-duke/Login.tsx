import React, { useReducer } from 'react';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import { initialLoginForm, loginReducer } from '@/reducer/loginReducer';

const login = () => {
  const [state, dispatch] = useReducer(loginReducer, initialLoginForm);
  return (
    <Container maxWidth="sm">
      <Typography variant="h4" component="h1" gutterBottom>
        IRON-DUKE
      </Typography>
      <Box sx={{ my: 4 }}>
        <form className={'login'} noValidate autoComplete="off">
          <TextField
            value={state.email}
            onChange={(e) => {
              dispatch({
                type: 'CHANGE_EMAIL',
                payload: e.target.value,
              });
            }}
            id="standard-basic"
            label="email"
            required
            margin="dense"
          />
          <TextField
            value={state.password}
            onChange={(e) => {
              dispatch({
                type: 'CHANGE_PASSWORD',
                payload: e.target.value,
              });
            }}
            id="outlined-basic"
            label="Password"
            variant="outlined"
            required
            margin="dense"
          />
          <div />
          <Button
            variant="contained"
            color="primary"
            size="large"
            className={'LoginButton'}
          >
            Login
          </Button>
        </form>
      </Box>
    </Container>
  );
};

export default login;
