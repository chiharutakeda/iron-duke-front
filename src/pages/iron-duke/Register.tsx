import React, { useReducer } from 'react';
import { gqlfetcher } from '@/fetch/fetcher';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import { initialUserForm, userRegistReducer } from '@/reducer/userRegistReducer';
import { createUser_M } from '@/gql/gql';

const login = () => {
  const [state, dispatch] = useReducer(userRegistReducer, initialUserForm);
  return (
    <Container maxWidth="sm">
      <Typography variant="h4" component="h1" gutterBottom>
        register
      </Typography>
      <form className={'register'} noValidate autoComplete="off">
        <TextField
          value={state.email}
          onChange={(e) => {
            dispatch({
              type: 'CHANGE_EMAIL',
              payload: e.target.value,
            });
          }}
          id="standard-basic"
          label="email"
          required
          margin="dense"
          fullWidth
        />
        <TextField
          value={state.password}
          onChange={(e) => {
            dispatch({
              type: 'CHANGE_PASSWORD',
              payload: e.target.value,
            });
          }}
          id="outlined-basic"
          label="Password"
          variant="outlined"
          required
          margin="dense"
          fullWidth
        />
        <div />
        <Button
          variant="contained"
          color="primary"
          size="large"
          className={'RegistButton'}
          onClick={() => {
            gqlfetcher(createUser_M,{ createUserUserData: state });
          }}
        >
          Regist
        </Button>
      </form>
    </Container>
  );
};

export default login;
