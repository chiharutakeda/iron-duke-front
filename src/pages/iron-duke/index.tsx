import * as React from 'react';
import Grid from '@material-ui/core/Grid';
import AllToDoList from '@/components/modules/AllToDoList';
import ToDoRegister from '@/components/modules/ToDoRegister';
import AllToDoListApollo from '@/components/modules/AllToDoListApollo';
import ToDoRegisterApollo from '@/components/modules/ToDoRegisterApollo';
import fetch from 'cross-fetch';
import { ApolloProvider } from '@apollo/client';
import { ApolloClient, InMemoryCache, HttpLink } from '@apollo/client';

export const client = new ApolloClient({
  uri: 'http://localhost:4000',
  cache: new InMemoryCache(),
  link: new HttpLink({ uri: 'http://localhost:4000/graphql', fetch }),
});

import { QueryClient, QueryClientProvider } from 'react-query';
import { ReactQueryDevtools } from 'react-query/devtools';
import AllToDoListReactQuery from '@/components/modules/AllToDoListReactQuery';
import ToDoRegisterReactQuery from '@/components/modules/ToDoRegisterReactQuery';
const queryClient = new QueryClient();

export default function Index() {
  return (
    <Grid container justifyContent="space-around">
      <Grid item xs={3}>
        <ToDoRegister />
        <AllToDoList />
      </Grid>

      <Grid item xs={3}>
        <ApolloProvider client={client}>
          <ToDoRegisterApollo />
          <AllToDoListApollo />
        </ApolloProvider>
      </Grid>

      <Grid item xs={3}>
        <QueryClientProvider client={queryClient}>
          <ReactQueryDevtools initialIsOpen={false} />
          <ToDoRegisterReactQuery />
          <AllToDoListReactQuery />
        </QueryClientProvider>
      </Grid>
    </Grid>
  );
}
