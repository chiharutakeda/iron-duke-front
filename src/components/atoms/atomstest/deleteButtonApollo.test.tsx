import ReactTestRenderer from 'react-test-renderer';
import DeleteButtonApollo from '../DeleteButtonApollo';
import { client } from '@/pages/iron-duke/index';
import { ApolloProvider } from '@apollo/client';
it('DeleteButtonApollo', async () => {
  const tree = await ReactTestRenderer.create(
    <ApolloProvider client={client}>
      <DeleteButtonApollo ID={1} />
    </ApolloProvider>
  ).toJSON();
  await expect(tree).toMatchSnapshot();
});
