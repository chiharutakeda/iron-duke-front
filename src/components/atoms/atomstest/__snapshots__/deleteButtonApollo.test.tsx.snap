// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButtonApollo 1`] = `
<button
  className="DeleteButton"
  onClick={[Function]}
>
  Delete
</button>
`;
